streaming/base/dataset.py

```diff
@@ -438,13 +440,27 @@ def __init__(self,
             for stream in streams:
                 stream.apply_default(default)
         else:
-            default = Stream(remote=remote,
-                             local=local,
-                             split=split,
-                             download_retry=download_retry,
-                             download_timeout=download_timeout,
-                             validate_hash=validate_hash,
-                             keep_zip=keep_zip)
+            kwargs = {
+                'remote': remote,
+                'local': local,
+                'split': split,
+                'download_retry': download_retry,
+                'download_timeout': download_timeout,
+                'validate_hash': validate_hash,
+                'keep_zip': keep_zip,
+                **kwargs,
+            }
+
+            # Construct a Stream instance using registry-based construction
+            default = construct_from_registry(
+                name='stream',
```

Comment:
maybe would be more clear to take a `stream_name` arg in the dataset constructor (with a default of `default_stream` or wtvr) and use that here? not a super strong opinion

Reply:
Yeah I think it's a good idea, makes it easier to specify which Stream class to use too